apps/admin-gui/src/app/shared/components/assign-groups-sponsored-members/assign-groups-sponsored-members-component.ts

```diff
@@ -0,0 +1,181 @@
+    private guiAuthResolver: GuiAuthResolver,
+    private attributesService: AttributesManagerService
+  ) {}
+
+  ngOnInit(): void {
+    this.submit.subscribe(() => this.handleGroups());
+    this.selection.changed.subscribe(() => this.isSubmitAllowed());
+    this.createGroupAuth = this.guiAuthResolver.isAuthorized('createGroup_Vo_Group_policy', [
+      { id: this.voId, beanName: 'Vo' },
+    ]);
+    this.attributesService.getVoAttributes(this.voId).subscribe((attributes) => {
+      this.manualMemberAddingBlocked = hasBooleanAttributeEnabled(
+        attributes,
+        Urns.VO_BLOCK_MANUAL_MEMBER_ADDING
+      );
+      this.groupsService
+        .getAllRichGroupsWithAttributesByNames(this.voId, this.groupAttrNames)
+        .subscribe((groups) => {
+          this.allVoGroups = groups.filter((grp) => grp.name !== 'members');
+          this.assignableGroups = this.filterAssignableGroups(groups);
+          this.isSubmitAllowed();
+        });
+    });
+  }
+
+  groupAssigmentChanged(): void {
+    this.selection.clear();
+
+    this.groupToCreate.name = '';
+    this.groupToCreate.description = '';
+    this.asSubgroup = false;
+    this.groupToCreate.parentGroupId = null;
+    this.isSubmitAllowed();
+  }
+
+  applyFilter(filterValue: string): void {
+    this.filterValue = filterValue;
+  }
+
+  onNameChange(name: string): void {
+    this.groupToCreate.name = name;
+    this.isSubmitAllowed();
+  }
+
+  onParentChange(parentGroup: Group): void {
+    if (parentGroup === null) {
+      this.groupToCreate.parentGroupId = null;
+    } else {
+      this.groupToCreate.parentGroupId = parentGroup.id;
+    }
+    this.isSubmitAllowed();
+  }
+
+  onAsSubgroupChange(asSubgroup: boolean): void {
+    this.asSubgroup = asSubgroup;
+    this.isSubmitAllowed();
+  }
+
+  onDescriptionChange(description: string): void {
+    this.groupToCreate.description = description;
+    this.isSubmitAllowed();
+  }
+
+  isSubmitAllowed(): void {
+    this.submitAllowed.emit(
+      !(
+        this.groupAssignment === null ||
+        (this.groupAssignment === 'existing' && this.selection.selected.length === 0) ||
+        (this.groupAssignment === 'new' &&
+          (this.groupToCreate.name.length === 0 || this.groupToCreate.description.length === 0)) ||
+        (this.asSubgroup && this.groupToCreate.parentGroupId === null)
+      )
+    );
+  }
+
+  handleGroups(): void {
+    this.loading = true;
+    if (this.groupAssignment === 'new') {
+      if (this.asSubgroup) {
+        this.groupsService
+          .createGroupWithParentGroupNameDescription(
+            this.groupToCreate.parentGroupId,
+            this.groupToCreate.name,
+            this.groupToCreate.description
+          )
+          .subscribe(
+            (group) => {
+              this.groupIds.push(group.id);
+              this.groupsToAdd.emit(this.groupIds);
+            },
+            () => (this.loading = false)
+          );
```

Comment:
Do not use the deprecated subscription variant in the new code, please (more places).
```suggestion
          .subscribe({
            next: (group) => {
              this.groupIds.push(group.id);
              this.groupsToAdd.emit(this.groupIds);
            },
            error: () => (this.loading = false)
          });
```